torch/csrc/jit/script/parser.h

```diff
@@ -314,27 +334,79 @@ struct Parser {
     L.expect(TK_IN);
     auto itrs = parseList(TK_NOTHING, ',', TK_NOTHING, &Parser::parseExp);
     L.expect(':');
-    auto body = parseStatements();
+    auto body = std::get<0>(parseStatements());
     return For::create(r, targets, itrs, body);
   }
-  TreeRef parseStatements() {
+
+  // First return value is the TreeRef to the list of statements we've parsed.
+  // The second one is an optional Decl for if we find type annotations in a
+  // comments
+  std::tuple<TreeRef, std::shared_ptr<TreeRef>> parseStatements() {
```

Comment:
This can be significantly cleaned up by changing it to `parseStatement(bool expect_indent=true)`

Then changing parseFunction to:

```
L.expect(TK_INDENT)
if(L.nextIf(TK_TYPE_COMMENT) {
   ....
}
parseStatements(/*expect_indent=*/false)
```
Currently it spends time both enabling type comments to appear where they should be able to appear, and then check that it doesn't happen.